File: core-scala/src/main/java/io/asterisque/core/ProtocolException.java
package io.asterisque.core;

import javax.annotation.Nonnull;
import javax.annotation.Nullable;

public class ProtocolException extends RuntimeException {
  public ProtocolException(@Nonnull String message) {
    super(message);
  }

  public ProtocolException(@Nonnull String message, @Nullable Throwable ex) {
    super(message, ex);
  }
}

File: core-scala/src/main/scala/io/asterisque/wire/ProtocolException.scala
package io.asterisque.wire

import javax.annotation.{Nonnull, Nullable}

class ProtocolException(@Nonnull message:String, @Nullable ex:Throwable) extends RuntimeException(message, ex) {
  def this(@Nonnull message:String) = this(message, null)
}
